File: backend/internal/model/notification.go
package model

import (
	"time"
)

type Notification struct {
	ID        int       `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID    int       `gorm:"not null;index" json:"user_id"`
	Type      string    `gorm:"size:50;not null" json:"type"`
	Title     string    `gorm:"size:200;not null" json:"title"`
	Content   string    `gorm:"type:text;not null" json:"content"`
	ActionURL string    `gorm:"size:500" json:"action_url"`
	Metadata  string    `gorm:"type:jsonb" json:"metadata"`
	IsRead    bool      `gorm:"default:false;index" json:"is_read"`
	CreatedAt time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt time.Time `gorm:"autoUpdateTime" json:"updated_at"`

	User User `gorm:"foreignKey:UserID" json:"user,omitempty"`
}

func (Notification) TableName() string {
	return "notifications"
}

const (
	TypeTaskAssigned        = "task.assigned"
	TypeTaskCompleted       = "task.completed"
	TypeTaskOverdue         = "task.overdue"
	TypeProjectCreated      = "project.created"
	TypeProjectCompleted    = "project.completed"
	TypeMemberAdded         = "member.added"
	TypeMemberRemoved       = "member.removed"
	TypeRoleChanged         = "role.changed"
	TypeOrganizationCreated = "organization.created"
	TypeCommentAdded        = "comment.added"
	TypeMention             = "mention"
	TypeSystem              = "system"
)

File: frontend/src/features/notifications/types/notification.ts
export interface Notification {
    id: number;
    user_id: number;
    type: string;
    title: string;
    content: string;
    action_url: string;
    metadata: Record<string, any>;
    is_read: boolean;
    created_at: string;
    updated_at: string;
}
 
export const NotificationTypes = {
    TASK_ASSIGNED: 'task.assigned',
    TASK_COMPLETED: 'task.completed',
    TASK_OVERDUE: 'task.overdue',
    PROJECT_CREATED: 'project.created',
    PROJECT_COMPLETED: 'project.completed',
    MEMBER_ADDED: 'member.added',
    MEMBER_REMOVED: 'member.removed',
    ROLE_CHANGED: 'role.changed',
    ORGANIZATION_CREATED: 'organization.created',
    COMMENT_ADDED: 'comment.added',
    MENTION: 'mention',
    SYSTEM: 'system',
     
    INVITATION_ACCEPTED: 'invitation.accepted',
    INVITATION_DECLINED: 'invitation.declined',
} as const;